python/ray/worker.py

```diff
@@ -468,6 +469,7 @@ def get_dashboard_url():
 """ray.node.Node: The global node object that is created by ray.init()."""
 
 
+@client_mode_hook
 def init(
```

Comment:
Shouldn't the user be required to use `ray.connect()`?

Reply:
That's a valid design option. The nice thing about doing this is, again, everywhere (including tests and examples and general hands-on experience) where folks expect `init()` to work locally, this provides that.